### ISSUE TITLE
Make the /containers/json API endpoint return a machine readable Status field

### ISSUE THREAD

**Issue by mariusGundersen** [docker/docker#18962]
Currently the [https://docs.docker.com/engine/reference/api/docker_remote_api_v1.21/#list-containers](`GET /containers/json`) endpoint returns a human readable `Status` field, which is useful for the `docker ps` cli, but not very useful for other API clients, since it's not machine readable. For example, this is what it has returned for me (simplified to only include the Id and Status fields from the returned JSON): 

```
Id                                                               Status
9080c3d4e54b1367cd659d5841953f0a0b0ed789d52805491ba752744ca7ed33 Created
da837d8058c56c0c038d4580f5f8c2197a7c383be7de106dfbc6d17849975f2a Exited (137) 12 days ago
624d6deaf83091659d563d2063c6fcf0275a4b945b02e5b2262557a64e6953f8 Up 8 minutes (Paused)
2a6e659fe99f33a5f43605e313fbee17c2eb91e317567d6fd98d69c5d30504db Up 5 seconds
```

It would be more useful if the Status field (or a new field, for backwards compatibility) returned the states from this state diagram:

[![State diagram from Remote API documentation](https://docs.docker.com/engine/reference/api/images/event_state.png)](https://docs.docker.com/engine/reference/api/docker_remote_api/#docker-events)

I therefore propose a new `State` field to the result of calling `GET /containers/json` which will have one of the values from that diagram, namely `created`, `running`, `paused` or `stopped` (I don't think it can ever have `deleted`, since, well, it's deleted).

**Comment by mariusGundersen** [comment]
After a bit of exploring in the code this looks like it should be pretty straight forward, since the State object already has a way to return [the state as a string](https://github.com/docker/docker/blob/master/container/state.go#L74), and all that is needed is to add a new State field to [the Container struct](https://github.com/docker/docker/blob/master/api/types/types.go#L135). The only problem with this is that `StateString` returns `exited` rather than `stopped`. 

It would also be useful to return the timestamp of the last state transition, but maybe this should be in a separate API

**Comment by thaJeztah** [comment]
I think it would be consistent with `docker inspect`, which returns;

```
    "State": {
        "Status": "running",
        "Running": true,
        "Paused": false,
        "Restarting": false,
        "OOMKilled": false,
        "Dead": false,
        "Pid": 12817,
        "ExitCode": 0,
        "Error": "",
        "StartedAt": "2015-12-24T02:49:18.036789166Z",
        "FinishedAt": "0001-01-01T00:00:00Z"
    },
```

(I just spotted the dodgy `FinishedAt`, we probably should return `nil` there, not sure)

**Comment by mariusGundersen** [comment]
Yes, and while looking into this I also noticed that the [https://docs.docker.com/engine/reference/api/docker_remote_api_v1.21/#list-containers](`/containers`) endpoint takes those values as filters, so it's very strange that it doesn't return them.

I think that the State object in `docker inspect` should return the last transition timestamp, rather than the startedAt/finishedAt timestamps, like I said in my previous comment. Or there should be a separate api endpoint for getting a snapshot of the state of all containers, in a similar format to what `docker events` returns (id, timestamp, state).

Anyways, I've created a small pull request for at least exposing the state of containers: #18966
